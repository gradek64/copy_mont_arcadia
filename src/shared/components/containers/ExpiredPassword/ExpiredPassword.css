/** @define ExpiredPassword */

.ExpiredPassword {
  margin: 10px auto 20px;
  padding: 0 $page-padding;
  max-width: 500px;
  &-form {
    padding: 0 $page-padding 10px;
  }
}

@media $min-tablet {
  .ExpiredPassword {
    &--signin {
      padding: 0 $page-padding;
      & .ExpiredPassword {
        &-form {
          padding: 0;
        }
      }
    }
  }
}

@media $min-laptop {
  .ExpiredPassword {
    &--signin {
      & .ExpiredPassword {
        &-accordion {
          width: 100%;
        }
      }
    }
  }
}