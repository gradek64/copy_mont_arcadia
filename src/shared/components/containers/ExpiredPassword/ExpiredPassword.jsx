import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { isFeatureDesktopResetPasswordEnabled } from '../../../selectors/featureSelectors'
import {
  forgetPwdRequest,
  resetPasswordLinkRequest,
  setForgetPassword,
} from '../../../actions/common/accountActions'
import {
  resetForm,
  setFormField,
  setFormMessage,
  touchedFormField,
} from '../../../actions/common/formActions'
import ForgetPasswordForm from '../../common/ForgetPasswordForm/ForgetPasswordForm'

class ExpiredPassword extends Component {
  static propTypes = {
    forgetPwd: PropTypes.bool.isRequired,
    forgetPasswordForm: PropTypes.shape({
      fields: PropTypes.shape({
        email: PropTypes.shape({
          value: PropTypes.string,
          isDirty: PropTypes.bool,
          isTouched: PropTypes.bool,
          isFocused: PropTypes.bool,
        }),
        isLoading: PropTypes.bool,
        message: PropTypes.oneOf([
          PropTypes.shape({}),
          PropTypes.shape({
            type: PropTypes.string,
            message: PropTypes.string,
          }),
        ]),
      }),
    }).isRequired,
    isFeatureDesktopResetPasswordEnabled: PropTypes.func.isRequired,
    forgetPwdRequest: PropTypes.func.isRequired,
    resetForm: PropTypes.func.isRequired,
    resetPasswordLinkRequest: PropTypes.func.isRequired,
    setForgetPassword: PropTypes.func.isRequired,
    setFormField: PropTypes.func.isRequired,
    setFormMessage: PropTypes.func.isRequired,
    touchedFormField: PropTypes.func.isRequired,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  render() {
    const { className, ...props } = this.props
    const { l } = this.context
    return (
      <section className="ExpiredPassword">
        <h2>{l`Reset link expired or not valid`}</h2>
        <p>{l`link no longer valid`}</p>
        <p>{l`link expires in one hour`}</p>
        <ForgetPasswordForm {...props} />
      </section>
    )
  }
}

export default connect(
  (state) => ({
    forgetPwd: state.account.forgetPwd,
    forgetPasswordForm: state.forms.forgetPassword,
    isFeatureDesktopResetPasswordEnabled: isFeatureDesktopResetPasswordEnabled(
      state
    ),
  }),
  {
    forgetPwdRequest,
    resetForm,
    resetPasswordLinkRequest,
    setForgetPassword,
    setFormField,
    setFormMessage,
    touchedFormField,
  }
)(ExpiredPassword)

export { ExpiredPassword as WrappedExpiredPassword }
