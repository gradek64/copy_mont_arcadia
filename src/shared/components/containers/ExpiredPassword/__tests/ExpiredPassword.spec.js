import testComponentHelper from '../../../../../../test/unit/helpers/test-component'
import { WrappedExpiredPassword as ExpiredPassword } from '../ExpiredPassword'

const renderComponent = testComponentHelper(ExpiredPassword)

describe('<ExpiredPassword />', () => {
  const props = {}

  beforeEach(() => {
    jest.resetAllMocks()
  })

  describe('@render', () => {
    it('renders', () => {
      expect(renderComponent(props).getTree()).toMatchSnapshot()
    })
  })
})
