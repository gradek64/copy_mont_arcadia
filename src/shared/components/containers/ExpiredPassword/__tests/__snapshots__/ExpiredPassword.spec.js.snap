// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpiredPassword /> @render renders 1`] = `
<section
  className="ExpiredPassword"
>
  <h2>
    Reset link expired or not valid
  </h2>
  <p>
    link no longer valid
  </p>
  <p>
    link expires in one hour
  </p>
  <Connect(ForgetPasswordForm) />
</section>
`;
